pkg/scheduler/framework/plugins/registry.go

```diff
@@ -59,52 +59,40 @@ func NewInTreeRegistry() runtime.Registry {
 		EnableGenericEphemeralVolume:       feature.DefaultFeatureGate.Enabled(features.GenericEphemeralVolume),
 	}
 
+	wrap := func(extendedFactory func(apiruntime.Object, framework.Handle, plfeature.Features) (framework.Plugin, error)) runtime.PluginFactory {
+		return func(plArgs apiruntime.Object, fh framework.Handle) (framework.Plugin, error) {
+			return extendedFactory(plArgs, fh, fts)
```

Comment:
not sure whether the signature of those plugins will be changed in the future, but seems like they are okay today.

/lgtm

Reply:
As some tests also use the old pattern extensively, I think we can make the `wrap` more reusable by not hard-wiring the `fts`; instead, add an `plfeature.Features` parameter to the function signature.

Here is my thought:

- add a function type in runtime/registry.go:

```go
type PluginFactoryWithFts func(runtime.Object, framework.Handle, plfeature.Features) (framework.Plugin, error)
``` 

- move the `wrap` as an "Adapter" function to runtime/registry.go, so it can be used elsewhere

```go
func FactoryAdapter(fts plfeature.Features, withFts PluginFactoryWithFts) PluginFactory {
	return func(plArgs runtime.Object, fh framework.Handle) (framework.Plugin, error) {
		return withFts(plArgs, fh, fts)
	}
}
```

- use the following pattern in all callers (plugins/registry.go and other tests)

```go
runtime.Adapter(fts, noderesources.NewFit)
runtime.Adapter(fts, noderesources.NewBalancedAllocation),
...
```

Reply:
Makes sense. I pushed an updated commit with that approach.

I've not tried updating other code because:
- I don't know where it is and initial "grep" wasn't very informative.
- It's likely to be in code with various owners, which would delay merging this PR if we need to get approval from many different SIGs.

Reply:
> I've not tried updating other code because

That's fair. I can follow up on that.